lib/services/s3-service.ts

```diff
@@ -0,0 +1,13 @@
+export class S3Service implements IS3Service {
```

Comment:
[Suggestion] Rename to a helper

Reply:
1. Done
2. I prefer not to move the httpRequest call from downloadServerResults in the s3 helper because we can have results not related to s3.